rts/Sim/Projectiles/ProjectileMemPool.h

```diff
@@ -5,11 +5,16 @@
 
 #include "Sim/Misc/GlobalConstants.h"
 #include "System/MemPoolTypes.h"
+#include "System/SpringMath.h"
+
+#include "Sim/Projectiles/WeaponProjectiles/StarburstProjectile.h"
+
+static constexpr size_t PMP_S = AlignUp(sizeof(CStarburstProjectile), 4); //biggest in size
```

Comment:
The magic constant was here to avoid including any specific projectile type header. If some other projectile type ever becomes the largest footprint you'll have to change both the `#include` and the `PMP_S` LOC so this really adds nothing over just keeping a raw number around.
A clean solution would be calculating max(sizeof(T1), sizeof(T1), ...) at compile-time, which also requires including every single projectile type header and just isn't worth it.

Reply:
Respectfully disagree. The day when some projectile will suddenly outgrow Starburst may never come, while the need to do guesswork of how big the magic constant should be will come right after a new field is added to CProjectile (or its base classes).